code/modules/food_and_drinks/recipes/soup_mixtures.dm

```diff
@@ -180,28 +207,25 @@
 
 	reaction?.data["ingredients"] = null
 
-	for(var/obj/item/ingredient as anything in pot.added_ingredients)
-		// Let's not mess with  indestructible items.
-		// Chef doesn't need more ways to delete things with cooking.
-		if(ingredient.resistance_flags & INDESTRUCTIBLE)
-			continue
-
-		// Things that had reagents or ingredients in the soup will get deleted
-		else if(!isnull(ingredient.reagents) || is_type_in_list(ingredient, required_ingredients))
-			LAZYREMOVE(pot.added_ingredients, ingredient)
-			// Send everything left behind
-			transfer_ingredient_reagents(ingredient, holder)
-			// Delete, it's done
-			qdel(ingredient)
+	// If soup is made, remove ingredients as their reagents were added to the soup
+	if(react_vol)
+		for(var/obj/item/ingredient as anything in pot.added_ingredients)
+			// Let's not mess with  indestructible items.
+			// Chef doesn't need more ways to delete things with cooking.
+			if(ingredient.resistance_flags & INDESTRUCTIBLE)
+				continue
 
-		// Everything else will just get fried
-		else
-			ingredient.AddElement(/datum/element/fried_item, 30)
+			// Things that had reagents or ingredients in the soup will get deleted
+			else if(!isnull(ingredient.reagents) || is_type_in_list(ingredient, required_ingredients))
+				LAZYREMOVE(pot.added_ingredients, ingredient)
+				// Send everything left behind
+				transfer_ingredient_reagents(ingredient, holder)
+				// Delete, it's done
+				qdel(ingredient)
```

Comment:
```suggestion
			if(isnull(ingredient.reagents) && !is_type_in_list(ingredient, required_ingredients))
				ingredient.AddElement(/datum/element/fried_item, 30)
				continue
			LAZYREMOVE(pot.added_ingredients, ingredient)
			// Send everything left behind
			transfer_ingredient_reagents(ingredient, holder)
			// Delete, it's done
			qdel(ingredient)
```

1. Don't need an else following continue
2. Let's get the shorter case out of the way first and return early

Reply:
yeah i was too lazy to switch to side-to-side im afraid, you'd have to make the change manually 😅

Reply:
Made changes, if your good with it, you can resolve this